typescript/src/redemption.ts

```diff
@@ -55,29 +56,36 @@ export interface RedemptionRequest {
 
 /**
  * Requests a redemption from the on-chain Bridge contract.
+ * @param redeemer - On-chain identifier of the redeemer.
  * @param walletPublicKey - The Bitcoin public key of the wallet. Must be in the
  *        compressed form (33 bytes long with 02 or 03 prefix).
- * @param mainUtxo - The main UTXO of the wallet. Must match the main UTXO
- *        held by the on-chain Bridge contract.
+ * @param mainUtxo - The main UTXO of the wallet. Must match the main UTXO held
+ *        by the on-chain Bridge contract.
  * @param redeemerOutputScript - The output script that the redeemed funds will
  *        be locked to. Must be un-prefixed and not prepended with length.
- * @param amount - The amount to be redeemed in satoshis.
- * @param bridge - Handle to the Bridge on-chain contract.
- * @returns Empty promise.
+ * @param amount - The amount to be redeemed with the precision of the tBTC
+ *        on-chain token contract.
+ * @param vault - The vault address.
+ * @param tBTCToken - Handle to the TBTCToken on-chain contract.
```

Comment:
I don't think we should pass more than one contract as a parameter. What if someone passes `TBTCToken` contract address but `DonationVault` instead of the `TBTCVault`? This doesn't make sense, but the API should limit the surface of developer errors.

Since the redemption happens on the token contract, I think we should pass `TBTCToken` contract as a parameter and resolve the vault [from the `owner()` call](https://github.com/thesis/solidity-contracts/blob/c315b9d5f0c39253428898474e97552f5e5b9046/contracts/token/ERC20WithPermit.sol#L18).

Reply:
06601b9